sdk/core/Azure.Core/src/Shared/AuthorizationChallengeParser.cs

```diff
@@ -27,17 +27,17 @@ internal static class AuthorizationChallengeParser
                 return null;
             }
 
-            ReadOnlySpan<char> bearer = challengeScheme.AsSpan();
-            ReadOnlySpan<char> claims = challengeParameter.AsSpan();
+            ReadOnlySpan<char> scheme = challengeScheme.AsSpan();
```

Comment:
Your PR should include only files that you're changing.   This does not belong here nor do many of the other changes that seem to have come in via merge commits.

Please rebase on `main` and ensure that only your intended changes are included.